.travis.yml

```diff
@@ -36,13 +37,13 @@ matrix:
           env: CONFIG_OPTS="-fsanitize=address"
         - os: linux
           compiler: clang-3.6
-          env: CONFIG_OPTS="--debug --strict-warnings -fsanitize=address"
+          env: CONFIG_OPTS="no-asm -DSTRICT_ALIGNMENT --debug --strict-warnings -fno-sanitize-recover -fsanitize=address -fsanitize=undefined"
         - os: linux
           compiler: gcc-5
           env: CONFIG_OPTS="-fsanitize=address"
         - os: linux
           compiler: gcc-5
-          env: CONFIG_OPTS="--debug --strict-warnings -fsanitize=address"
+          env: CONFIG_OPTS="no-asm -DSTRICT_ALIGNMENT --debug --strict-warnings -fno-sanitize-recover -fsanitize=address -fsanitize=undefined"
```

Comment:
Well, problem with solving it this way, i.e. passing STRICT_ALIGNMENT as command-line option, is that the macro is used as *internal* in several other places. And if one chooses to define it at command line, one has to answer questions about how does it affect those other files. For this reason I'd suggest to abstain from defining it at command line and solve it rather as following

```diff
diff --git a/crypto/whrlpool/wp_block.c b/crypto/whrlpool/wp_block.c
index 7baca60..7e49383 100644
--- a/crypto/whrlpool/wp_block.c
+++ b/crypto/whrlpool/wp_block.c
@@ -50,9 +50,10 @@ typedef unsigned long long u64;
 #define ROUNDS  10
 
 #define STRICT_ALIGNMENT
-#if defined(__i386) || defined(__i386__) || \
-    defined(__x86_64) || defined(__x86_64__) || \
-    defined(_M_IX86) || defined(_M_AMD64) || defined(_M_X64)
+#if !defined(PEDANTIC) && (defined(__i386) || defined(__i386__) || \
+                           defined(__x86_64) || defined(__x86_64__) || \
+                           defined(_M_IX86) || defined(_M_AMD64) || \
+                           defined(_M_X64))
 /*
  * Well, formally there're couple of other architectures, which permit
  * unaligned loads, specifically those not crossing cache lines, IA-64 and
@@ -82,17 +83,18 @@ typedef unsigned long long u64;
 #endif
 
 #undef ROTATE
-#if defined(_MSC_VER)
-# if defined(_WIN64)            /* applies to both IA-64 and AMD64 */
-#  pragma intrinsic(_rotl64)
-#  define ROTATE(a,n) _rotl64((a),n)
-# endif
-#elif defined(__GNUC__) && __GNUC__>=2
-# if defined(__x86_64) || defined(__x86_64__)
-#  if defined(L_ENDIAN)
-#   define ROTATE(a,n)       ({ u64 ret; asm ("rolq %1,%0"   \
+#ifndef PEDANTIC
+# if defined(_MSC_VER)
+#  if defined(_WIN64)            /* applies to both IA-64 and AMD64 */
+#   pragma intrinsic(_rotl64)
+#   define ROTATE(a,n) _rotl64((a),n)
+#  endif
+# elif defined(__GNUC__) && __GNUC__>=2
+#  if defined(__x86_64) || defined(__x86_64__)
+#   if defined(L_ENDIAN)
+#    define ROTATE(a,n)       ({ u64 ret; asm ("rolq %1,%0"   \
                                    : "=r"(ret) : "J"(n),"0"(a) : "cc"); ret; })
-#  elif defined(B_ENDIAN)
+#   elif defined(B_ENDIAN)
        /*
         * Most will argue that x86_64 is always little-endian. Well, yes, but
         * then we have stratus.com who has modified gcc to "emulate"
@@ -100,16 +102,17 @@ typedef unsigned long long u64;
         * won't do same for x86_64? Naturally no. And this line is waiting
         * ready for that brave soul:-)
         */
-#   define ROTATE(a,n)       ({ u64 ret; asm ("rorq %1,%0"   \
+#    define ROTATE(a,n)       ({ u64 ret; asm ("rorq %1,%0"   \
                                    : "=r"(ret) : "J"(n),"0"(a) : "cc"); ret; })
-#  endif
-# elif defined(__ia64) || defined(__ia64__)
-#  if defined(L_ENDIAN)
-#   define ROTATE(a,n)       ({ u64 ret; asm ("shrp %0=%1,%1,%2"     \
+#   endif
+#  elif defined(__ia64) || defined(__ia64__)
+#   if defined(L_ENDIAN)
+#    define ROTATE(a,n)       ({ u64 ret; asm ("shrp %0=%1,%1,%2"     \
                                    : "=r"(ret) : "r"(a),"M"(64-(n))); ret; })
-#  elif defined(B_ENDIAN)
-#   define ROTATE(a,n)       ({ u64 ret; asm ("shrp %0=%1,%1,%2"     \
+#   elif defined(B_ENDIAN)
+#    define ROTATE(a,n)       ({ u64 ret; asm ("shrp %0=%1,%1,%2"     \
                                    : "=r"(ret) : "r"(a),"M"(n)); ret; })
+#   endif
 #  endif
 # endif
 #endif
```
As for ROTATE section missing OPENSSL_NO_ASM, but favoring PEDANTIC. This is based so to say on precedent. In other words idea is to harmonize it with other similar cases, where we have both PEDANTIC and OPENSSL_NO_ASM (and even OPENSSL_NO_INLINE_ASM), *with PEDANTIC being outermost*. Second part of "precedent" rationale behind not introducing OPENSSL_NO_ASM is that there is no OPENSSL_NO_ASM in wp_block.c and it worked fine so far. My understanding is that original purpose of NO_ASM was to provide a way to  disengage inline assembly code in cases when there is suspicion that there is a problem caused by compiler bug. As no problem were reported so far, there really is little reason to change it more that minimally required. To summarize, outermost PEDANTIC is favoured because that's how it was done so far in other similar cases, and NO_ASM is not added because it's not actually required to solve this particular problem. One can argue that it's a twisted argument, and wp_block.c deserves a makeover (along originally suggested lines) anyway. But then one would have to discuss more things deserving makeover. But then it's more appropriate to solve this problem with minimally required changes, and discuss makeovers separately...